native/src/modules/endpoint/reducers/pushOrRefreshCategory.js

```diff
@@ -0,0 +1,149 @@
+    ? { ...state.resourceCache.value, ...resourceCache }
+    : resourceCache
+
+  const root: ?CategoryModel = categoriesMap.findCategoryByPath(path)
+
+  if (!root) {
+    const route: CategoryRouteStateType = {
+      path: path,
+      depth: depth,
+      language,
+      city,
+      status: 'error',
+      message: `Could not find a category with path '${path}'.`,
+      code: ErrorCodes.PageNotFound
+    }
+
+    return {
+      ...state,
+      categoriesRouteMapping: {
+        ...state.categoriesRouteMapping,
+        [key]: route
+      },
+      resourceCache: {
+        status: 'ready',
+        value: newResourceCache
+      },
+      searchRoute: { categoriesMap }
+    }
+  }
+
+  // Check whether another page in the same city is loading, e.g. because it is being refreshed.
+  // This is important for displaying the loading spinner.
+  const otherPageLoading = Object.values(state.categoriesRouteMapping)
+    // $FlowFixMe Flow does not support Object.values
+    .filter(route => city === route.city && path !== route.path && language === route.language)
+    // $FlowFixMe Flow does not support Object.values
+    .some(route => route.status === 'loading')
+
+  const { resultModels, resultChildren } = extractResultModelsAndChildren(root, categoriesMap, depth)
+
+  // $FlowFixMe Flow can't evaluate the status as it is dynamic
+  const route: CategoryRouteStateType = {
+    path: root.path,
+    models: resultModels,
+    children: resultChildren,
+    depth: depth,
+    allAvailableLanguages: getAllAvailableLanguages(root, city, language, cityLanguages),
+    language,
+    city,
+    status: (otherPageLoading && type === 'PUSH_CATEGORY') ? 'loading' : 'ready'
+  }
+
+  if (type === 'REFRESH_CATEGORY') {
+    // Update all open routes in the same city with the new content in case the content has been refreshed
+    Object.entries(state.categoriesRouteMapping)
+      // $FlowFixMe Object.entries does not supply proper types
+      .filter(([key, route]) => city === route.city && path !== route.path && language === route.language)
+      .forEach(([key: string, route]) => {
+        // $FlowFixMe Object.entries does not supply proper types
+        const root: ?CategoryModel = categoriesMap.findCategoryByPath(route.path)
+        if (!root) {
+          // $FlowFixMe Object.entries does not supply proper types
+          state.categoriesRouteMapping[key] = {
+            path: path,
+            depth: depth,
+            language,
+            city,
+            status: 'error',
+            message: `Could not find a category with path '${path}'.`,
+            code: ErrorCodes.PageNotFound
+          }
+        } else {
+          const { resultModels, resultChildren } = extractResultModelsAndChildren(root, categoriesMap, depth)
+          // $FlowFixMe Object.entries does not supply proper types
+          state.categoriesRouteMapping[key] = {
+            // $FlowFixMe Object.entries does not supply proper types
```

Comment:
The next line does not contain an Object.entries

If there are that many flowfixmes it makes sense to refactor this or implement the function yourself.

Reply:
You can also extract it to a separate function and handle the flowfixme only once.